src/modal-v2/modal-v2.tsx

```diff
@@ -0,0 +1,148 @@
+    // CONST, STATE, REF
+    // =============================================================================
+    const [verticalHeight, setVerticalHeight] = useState<number>();
+    const [offsetTop, setOffsetTop] = useState<number>();
+
+    const { refs, context } = useFloating({
+        open: show,
+        onOpenChange: (isOpen) => {
+            if (!isOpen) {
+                onClose();
+            }
+        },
+    });
+    const { isMounted, status } = useTransitionStatus(context, {
+        duration: 300,
+    });
+
+    const dismiss = useDismiss(context, {
+        /* handled by overlayclick */
+        outsidePress: false,
+    });
+    const { getFloatingProps } = useInteractions([dismiss]);
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        // set initial vh
+
+        // use VisualViewport API if available, it gives more accurate dimensions when iOS software keyboard is active
+        if (window.visualViewport) {
+            handleViewportResize();
+            window.visualViewport.addEventListener(
+                "resize",
+                handleViewportResize
+            );
+            return () => {
+                window.visualViewport?.removeEventListener(
+                    "resize",
+                    handleViewportResize
+                );
+            };
+        } else {
+            // fallback to Window API
+            handleWindowResize();
+            window.addEventListener("resize", handleWindowResize);
+            return () => {
+                window.removeEventListener("resize", handleWindowResize);
+            };
+        }
+    }, []);
+
+    useEffect(() => {
+        if (show && dismissKeyboardOnShow) {
+            // dismiss software keyboard to put modal in fullscreen
+            (document.activeElement as HTMLElement)?.blur?.();
+        }
+    }, [dismissKeyboardOnShow, show]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleWindowResize = () => {
+        const newVerticalHeight = window.innerHeight * 0.01;
+        setVerticalHeight(newVerticalHeight);
+    };
+
+    const handleViewportResize = () => {
+        if (window.visualViewport) {
+            const newVerticalHeight = window.visualViewport.height * 0.01;
+            setVerticalHeight(newVerticalHeight);
+            setOffsetTop(window.visualViewport.offsetTop);
+        }
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+
+    return (
+        <Overlay
+            data-testid={`${id}-overlay`}
```

Comment:
destructure `"data-testid": testId = "modal",`, set the data-testid to `${testId}-overlay`